pkg/profiler/cpu/maps.go

```diff
@@ -82,38 +84,81 @@ type bpfMaps struct {
 }
 
 func initializeMaps(m *bpf.Module, byteOrder binary.ByteOrder) (*bpfMaps, error) {
-	debugPIDs, err := m.GetMap(debugPIDsMapName)
+	if m == nil {
+		return nil, fmt.Errorf("nil module")
+	}
+
+	maps := &bpfMaps{
+		module:    m,
+		byteOrder: byteOrder,
+	}
+
+	return maps, nil
+}
+
+// adjustMapSizes updates unwinding maps' maximum size. By default, it tries to keep it as low
+// as possible.
+//
+// Note: It must be called before `BPFLoadObject()`.
+func (m *bpfMaps) adjustMapSizes(enableDWARFUnwinding bool) error {
+	unwindTables, err := m.module.GetMap(unwindTablesMapName)
 	if err != nil {
-		return nil, fmt.Errorf("get debug pids map: %w", err)
+		return fmt.Errorf("get unwind tables map: %w", err)
 	}
 
-	stackCounts, err := m.GetMap(stackCountsMapName)
+	dwarfStackTraces, err := m.module.GetMap(dwarfStackTracesMapName)
 	if err != nil {
-		return nil, fmt.Errorf("get counts map: %w", err)
+		return fmt.Errorf("get dwarf stack traces map: %w", err)
+	}
+
+	// Adjust unwind tables size.
+	if enableDWARFUnwinding {
+		sizeBefore := unwindTables.GetMaxEntries()
+		if err := unwindTables.Resize(unwindTableMaxEntries); err != nil {
+			return fmt.Errorf("resize unwind tables map from %d to %d elements: %w", sizeBefore, unwindTableMaxEntries, err)
+		}
+
+		sizeBefore = dwarfStackTraces.GetMaxEntries()
+		if err := dwarfStackTraces.Resize(maxStackTraceCount); err != nil {
+			return fmt.Errorf("resize dwarf stack traces map from %d to %d elements: %w", sizeBefore, maxStackTraceCount, err)
+		}
 	}
 
-	stackTraces, err := m.GetMap(stackTracesMapName)
+	return nil
+}
+
+func (m *bpfMaps) load() error {
```

Comment:
nit: we are creating them, perhaps `create` is a better verb? (we typically load programs, but create maps)